R/designOfExperiments.R

```diff
@@ -17,4 +17,188 @@
 
 designOfExperiments <- function(jaspResults, dataset, options, ...){
 
+  .qualityControlShowAvailableDesigns(options, jaspResults, position = 1)
+
+  .qualityControlSummarySelectedDesign(options, jaspResults, position = 2)
+
+  .qualityControlShowSelectedDesign(options, jaspResults, position = 3)
+
+}
+
+.qualityControlShowAvailableDesigns <- function(options, jaspResults, position){
+
+  if(is.null(jaspResults[["displayDesigns"]])){
+
+    tableTitle <- base::switch(options[["design"]],
+                               "factorial" = gettext("Available Factorial Designs (with Resolution)"),
+                               "screening" = gettext("Available Screening Designs"),
+                               "response" = gettext("Available Response Surface Designs"),
+                               "mixture" = gettext("Available Mixture Designs"),
+                               "taguchi" = gettext("Available Taguchi Designs"))
+
+    table <- createJaspTable(tableTitle)
+    table$position <- position
+
+    table$dependOn(options = "design")
+
+    if(options[["design"]] == "factorial"){
+
+      table$addColumnInfo(name = 'run', 		title = gettext("Runs"), 	type = 'integer')
+      table$addColumnInfo(name = 'factor2', 	title = "2", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor3', 	title = "3", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor4', 	title = "4", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor5', 	title = "5", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor6', 	title = "6", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor7', 	title = "7", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor8', 	title = "8", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor9', 	title = "9", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor10', 	title = "10", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor11', 	title = "11", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor12', 	title = "12", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor13', 	title = "13", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor14', 	title = "14", 				type = 'string', overtitle = "Factors")
+      table$addColumnInfo(name = 'factor15', 	title = "15", 				type = 'string', overtitle = "Factors")
```

Comment:
```suggestion
      for (i in 2:15) {
        table$addColumnInfo(name = paste0("factor", i), title = as.character(i), type = "string", overtitle = "Factors")
}
```
also mixing `"` and `'`